### DISCUSSION TITLE
Use Swift SDK with objc language

### DISCUSSION THREAD

**Discussion by MikoShao** [WalletConnect/walletconnect-monorepo#1011] [Developer Support]
Can I use this library in an objc project? anyone provide some advice and help on how to do it? 

I want to compile this library into a game project, but the game projects are all in oc language.

**Comment by dukevantreta** [comment]
Yes, Obj-C and Swift are completely interoperable in both ways. You will just need a bridging header to forward declare the Swift interfaces and use them as if they were Obj-C code.

Here's some info on how to do it: https://developer.apple.com/documentation/swift/imported_c_and_objective-c_apis/importing_swift_into_objective-c

**Comment by MikoShao** [comment]
@MisterVants
Yes, I have checked a lot of related technical articles, objc requires swift's class annotation @objc and public, I'm still trying, albeit with some difficulties.

At present, an error occurs when running ExampleApp, as follows, is there a problem with the server?

```
WebSocket Error receiveMessageFailure
(Error Domain=NSPOSIXErrorDomain 
Code=57 "Socket is not connected" 
UserInfo={NSErrorFailingURLStringKey=https://relay.dev.walletconnect.com/?projectId=52af113ee0c1e1a20f4995730196c13e, 
NSErrorFailingURLKey=https://relay.dev.walletconnect.com/?projectId=52af113ee0c1e1a20f4995730196c13e}
)
```
And I am more concerned about when the official version of WalletConnectV2 will be released?

**Comment by Edouardfrancois** [comment]
hello, I have the same problem. If you had resolved it can you help me plz ?
